raiden/messages.py

```diff
@@ -610,23 +647,24 @@ def pack(self, packed) -> None:
 
 @dataclass(repr=False, eq=False)
 class LockedTransfer(LockedTransferBase):
-    """
-    A LockedTransfer has a `target` address to which a chain of transfers shall
-    be established. Here the `secrethash` is mandatory.
-
-    `fee` is the remaining fee a recipient shall use to complete the mediated transfer.
-    The recipient can deduct his own fee from the amount and lower `fee` to the remaining fee.
-    Just as the recipient can fail to forward at all, or the assumed amount,
-    it can deduct a too high fee, but this would render completion of the transfer unlikely.
-
-    The initiator of a mediated transfer will calculate fees based on the likely fees along the
-    path. Note, it can not determine the path, as it does not know which nodes are available.
-
-    Initial `amount` should be expected received amount + fees.
-
-    Fees are always payable by the initiator.
-
-    `initiator` is the party that knows the secret to the `secrethash`
+    """ Message to used to reserve tokens for a new mediated transfer.
+
+    For this message to be valid, the sender must:
+
+    - Use a lock.amount smaller then its current capacity. If the amount is
+      higher then the recipient will be reject it, as it means spending money
+      it does not own.
+    - Have the new lock represented in merkleroot.
+    - Increase the locked_amount by exactly `lock.amount` otherwise the message
+      would be rejected by the recipient. If the locked_amount is increase by
+      more, then funds may get locked in the channel. If the locked_amount is
```

Comment:
> then funds may get locked in the channel.

I don't follow this. 
1. If the locked amount is bigger than the node's capacity in a channel, the partner will simply reject the transfer in the first place.
2. Even if  we end up in that situation, a transfer's lock will expire regardless of it's amount right? this line seems like the locked amount will be locked indefinitely.

Reply:
> If the locked amount is bigger than the node's capacity in a channel, the partner will simply reject the transfer in the first place.

In our implementation because the nodes are well behave that is true, but there is no incentive for this to be the case, from the receiver's perspective it's just an over commitment and he has nothing to lose.

> Even if we end up in that situation, a transfer's lock will expire regardless of it's amount right? this line seems like the locked amount will be locked indefinitely

It will, because the receiver won't allow the locked_amount to be decreased by an amount larger then `lock.amount`, otherwise the sender would be under committing. 

The tokens are only returned back after unlock, however there is a bad edge case when the merkle tree is empty [here](https://github.com/raiden-network/raiden-contracts/blob/7fb9fcf0575a801bc438b868120746fc7da773b8/raiden_contracts/data/source/raiden/TokenNetwork.sol#L807)